### PR TITLE
Feature/milestone 3

### PR DESCRIPTION
**Pull request by oscardelalanza**
### What was made
- `BooksFrom` is stateful
- `handleChange` method added to `BooksForm`
- `handleSubmit` method added to `BooksForm`
- `removeBook` action connected to `BooksList` and passed as a `prop` to `Book` component

### CODE REVIEW

File: src/containers/BooksForm.jsx [hunk 1]
```diff
@@ -1,27 +1,80 @@
...
-            <option selected>Select Category</option>
-            {categories.map(cat => <option key={cat} value={cat}>{cat}</option>)}
-          </select>
-        </label>
+        <h3>Add New Book</h3>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <label htmlFor="title">
+              Title:
+              <input type="text" id="title" name="title" placeholder="Book Title" onChange={this.handleChange} value={title} />
```
Comment (suretrust) [inline src/containers/BooksForm.jsx]: You could do the validation like so.
```suggestion
              <input type="text" id="title" name="title" placeholder="Book Title" onChange={this.handleChange} value={title} required />
```

File: src/containers/BooksForm.jsx [hunk 2]
```diff
@@ -1,27 +1,80 @@
...
+            </label>
+          </div>
+          <div>
+            <label htmlFor="category">
+              Category:
+              <select id="category" name="category" value={category} onChange={this.handleChange}>
```
Comment (suretrust) [inline src/containers/BooksForm.jsx]: And this. See how it works [here](https://www.w3schools.com/tags/att_select_required.asp).
```suggestion
              <select id="category" name="category" value={category} onChange={this.handleChange} required>
```

File: src/components/Book.jsx [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
 
-const Book = ({ book }) => (
-  <tr>
-    <td>{book.id}</td>
-    <td>{book.title}</td>
-    <td>{book.category}</td>
-  </tr>
-);
+const Book = ({ book, removeBook }) => {
+  const handleRemove = () => removeBook(book);
```
Comment (suretrust) [inline src/components/Book.jsx:5]: This should be implemented in the `BookList` component. Then pass the `handleRemove` to `removeBook` props of the `Book` component.
```suggestion
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

**Comment by oscardelalanza** [comment]
### Changes Made
- validations added to `input` and `select` elements.
- `handleRemove` modified to be implemented in `BooksList`.
- `preventDefalt` moved to first line.

File: src/reducers/books.js [hunk 4]
```diff
@@ -3,14 +3,10 @@ import { CREATE_BOOK, REMOVE_BOOK } from '../actions';
 const bookReducer = (state = {}, action) => {
   switch (action.type) {
     case CREATE_BOOK:
-      return {
-        books: state.books.concat(action.book),
-      };
+      return [...state].concat(action.book);
```
Comment (Torres-ssf) [inline src/reducers/books.js]: **Optional**
- [x] You can simplify this line by changing it to:
```javascript
return [...state, action.book];
```

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi @oscardelalanza :hand: 
Great job on this milestone :clap: 
Good luck with the next one :smile: 

Happy Coding :computer: 
**Sergio Torres**

**Pull request merged**